core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala

```diff
@@ -74,176 +84,71 @@ class MultiCloudWorkspaceServiceSpec extends AnyFlatSpec with Matchers with Opti
-      )(testContext)
-    )
-    val workspaceRequest = WorkspaceRequest(
-      "test-bp-name",
-      UUID.randomUUID().toString,
-      Map.empty,
-      None,
-      None,
-      None,
-      None
-    )
-    val workspaceService = mock[WorkspaceService](RETURNS_SMART_NULLS)
-
-    doReturn(Future.successful(testData.azureBillingProject))
-      .when(mcWorkspaceService)
-      .getBillingProjectContext(any(), any())
-
-    doReturn(Some(testData.azureBillingProfile))
-      .when(bpDAO)
-      .getBillingProfile(any[UUID], any[RawlsRequestContext])
-
-    val result = Await.result(mcWorkspaceService.createMultiCloudOrRawlsWorkspace(
-                                workspaceRequest,
-                                workspaceService
-                              ),
-                              Duration.Inf
-    )
-
-    result.workspaceType shouldBe Some(WorkspaceType.McWorkspace)
-    result.cloudPlatform shouldBe Some(WorkspaceCloudPlatform.Azure)
-  }
-
   it should "return forbidden if creating a workspace against a billing project that the user does not have the createWorkspace action for" in {
-    val workspaceManagerDAO = new MockWorkspaceManagerDAO()
-    val config = MultiCloudWorkspaceConfig(ConfigFactory.load())
-    val samDAO = Mockito.spy(new MockSamDAO(slickDataSource))
-    val bpDAO = mock[BillingProfileManagerDAO](RETURNS_SMART_NULLS);
-    val leonardoDAO: LeonardoDAO = new MockLeonardoDAO()
-
+    val samDAO = mock[SamDAO]
     when(
       samDAO.userHasAction(SamResourceTypeNames.billingProject,
-                           "fake_mc_billing_project_name",
-                           SamBillingProjectActions.createWorkspace,
-                           testContext
+        testData.azureBillingProject.projectName.value,
+        SamBillingProjectActions.createWorkspace,
+        testContext
       )
     ).thenReturn(Future.successful(false))
-    val mcWorkspaceService = spy(
-      MultiCloudWorkspaceService.constructor(
-        slickDataSource,
-        workspaceManagerDAO,
-        bpDAO,
-        samDAO,
-        config,
-        leonardoDAO,
-        workbenchMetricBaseName
-      )(testContext)
+    val billingRepository = mock[BillingRepository]
+    when(billingRepository.getBillingProject(any())).thenReturn(Future(Some(testData.azureBillingProject)))
+    val bpmDAO = mock[BillingProfileManagerDAO]
+    when(bpmDAO.getBillingProfile(any(), any())).thenReturn(Some(testData.azureBillingProfile))
+    val service = new MultiCloudWorkspaceService(
+      testContext,
+      mock[WorkspaceManagerDAO],
+      bpmDAO,
+      samDAO,
+      mock[MultiCloudWorkspaceConfig],
+      mock[LeonardoDAO],
+      "MultiCloudWorkspaceService-test",
+      mock[WorkspaceManagerResourceMonitorRecordDao],
+      mock[WorkspaceRepository],
+      billingRepository
     )
     val workspaceRequest = WorkspaceRequest(
-      "fake_mc_billing_project_name",
+      testData.azureBillingProject.projectName.value,
```

Comment:
This is a more realistic test, since now we're checking permissions for the same project we're returning.